src/main/java/org/tikv/common/util/ChannelFactory.java

```diff
@@ -89,9 +89,15 @@ public CertWatcher(long pollInterval, List<File> targets, Runnable onChange) {
           this::tryReload, pollInterval, pollInterval, TimeUnit.SECONDS);
     }
 
+    // If any execution of the task encounters an exception, subsequent executions are suppressed.
     private void tryReload() {
-      if (needReload()) {
-        onChange.run();
+      // Add exception handling to avoid schedule stop.
+      try {
+        if (needReload()) {
+          onChange.run();
+        }
+      } catch (Exception e) {
```

Comment:
We might add a unit test for this behaviour.

Reply:
@iosmanthus Thanks for your review. I will add a unit test.